scenarios/kubernetes_verify.py

```diff
@@ -98,12 +108,58 @@ def main(branch, script, force, on_prow):
             'bash', '-c', 'cd kubernetes && %s' % script,
         )
     else:
-        os.environ["REPO_DIR"] = k8s # hack/lib/swagger.sh depends on this
-        os.environ["KUBE_FORCE_VERIFY_CHECKS"] = str(force)
-        os.environ["KUBE_VERIFY_GIT_BRANCH"] = str(branch)
-        check(
-            'bash', '-c', 'cd kubernetes && %s' % script,
-        )
+        # move checkout to correct path
+        k8s = '/go/src/k8s.io/kubernetes'
+        check('bash', '-c', 'file %s' % k8s)
+        #check('bash', '-c', 'mkdir -p %s' % k8s)
+        #check('bash', '-c', 'mv . %s' % k8s)
+        # make sure we are cd-ed to the repo
+        os.chdir(k8s)
+        # setup env
+        os.environ['REPO_DIR'] = k8s # hack/lib/swagger.sh depends on this
+        os.environ['KUBE_FORCE_VERIFY_CHECKS'] = str(force)
+        os.environ['KUBE_VERIFY_GIT_BRANCH'] = str(branch)
+        # TODO(bentheelder): mimic the other script too, and eventually
+        # migrate this flag to be a bool between the two options instead
+        if script == './hack/jenkins/verify-dockerized.sh':
+            # mimic k8s.io/kubernetes/hack/jenkins/verify-dockerized.sh
+            # do screwy hack/jenkins things / mimic the script
+            path_prefix = os.environ['GOPATH'] + '/bin'
+            path_prefix += ':' + os.getcwd() + '/third_party/etcd:/usr/local/go/bin'
+            os.environ['PATH'] = path_prefix + ':' + os.environ['PATH']
+            os.environ['ARTIFACTS_DIR'] = os.environ['WORKSPACE'] + '/artifacts'
+            os.environ['LOG_LEVEL'] = '4'
+
+            # TODO(BenTheElder): remove this horrible debug hack
+            # patch build/common to properly not rsync in .git
+            check('bash', '-c',
+                  'sed -i \'s#/.git/#/.git#\' ./build/common.sh && cat ./build/common.sh')
+            # make a dummy git commit so that the working tree is "clean"
+            # # as far as the scripts are concerned
+            check('bash', '-c', 'git add . && git commit -m "dummy commit"')
+            # end horrible hack
```

Comment:
The snippet above is an awful, terrible, no-good hack; but it did prove that this works. This patches out a bug in the upstream rsync, build, data container circus related to filtering out .git when rsync-ing into the build container.